### PR TITLE
Homework2

### PR DESCRIPTION
**Pull request by MonikaZelechowska**

### CODE REVIEW

File: module2/homework/generateSequence/generateSequence.hpp [hunk 1]
```diff
@@ -2,6 +2,9 @@
 #include <vector>
 
 std::vector<int> generateSequence(int count, int step) {
-    // TODO: Implement me :)
-    return {};
+    std::vector<int> generatedVec {};
```
Comment (piezol) [inline module2/homework/generateSequence/generateSequence.hpp:5]: We know the desired size of vector upfront so we should allocate memory in constructor or by using reserve() function.

File: module2/homework/nwdNww/nwdNww.hpp [hunk 2]
```diff
@@ -1,11 +1,18 @@
...
+            max_divider = i;
+    }
+    return max_divider;
 }
 
 int NWW(int lhs, int rhs) {
-    // TODO: Implement me :)
-    return -1;
+    if (NWD(lhs, rhs)!= 0)
+        return abs(lhs*rhs)/NWD(lhs, rhs);
```
Comment (piezol) [inline module2/homework/nwdNww/nwdNww.hpp:15]: I'm copying my comment from different PR:
There are trivial cases that your code won't work for, ie NWW(1e8, 1e8).
That is due to lhs*rhs overflowing int.
You can fix it by casting lhs to unsigned long long in the multiplication.

There are some nontrivial cases that will still fail, ie NWW(1e8-1, 1e8+1), but they fail for the std:lcm(int, int) as well, so it's ok.

File: module2/homework/nwdNww/nwdNww.hpp [hunk 3]
```diff
@@ -1,11 +1,18 @@
...
+    else 
+        return abs(lhs*rhs);
```
Comment (piezol) [inline module2/homework/nwdNww/nwdNww.hpp:17]: Which would always be equal to 0, so we can return 0 :)

File: module2/homework/nwdNww/nwdNww.hpp (hunk 2 above)
Comment (piezol) [reply module2/homework/nwdNww/nwdNww.hpp:15]: Also, std::abs instead of abs

**Comment by ziobron** [comment]
{
"requirement_ids": [1119, 1631, 1119, 1633, 1119, 1634, 1119, 1635],
"points": [2, 4, 2, 6, 2, 5, 2, 5]
}

**Pull request closed without merging**